tests/test_gates_gates.py

```diff
@@ -747,7 +758,7 @@ def test_fswap(backend):
             [0, 1, 0, 0],
             [0, 0, 0, -1],
         ],
-        dtype=backend.dtype,
+        dtype=np.complex128,
```

Comment:
What is the reason why we are changing this everywhere? In principle it should be safer to rely on the `backend.dtype`, right?

Reply:
If the backend is pytorch the backend.dtype will be torch.dtype which can't be used in the function np.array